subsys/jaxrs/src/main/java/org/commonjava/indy/bind/jaxrs/RequestContextConstants.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2011-2018 Red Hat, Inc. (https://github.com/Commonjava/indy)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.bind.jaxrs;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+import static org.commonjava.indy.IndyRequestConstants.HEADER_COMPONENT_ID;
+
+/**
+ * The scope annotations (Thread, Header, MDC) tell where the constant is available/used.
+ */
+public class RequestContextConstants
+{
+    // Scope annotations
+
+    @Target( { FIELD } )
```

Comment:
Are these annotations for later use? That's a nice idea.

Reply:
They are used in this exact class. I once thought of using comment but later I thought anno is more clear. I want anyone reading this code be able to figure out which can be used in logback/MDC, which is in thread context, which can be found in request header.